modules/stochastic_tools/src/controls/MultiAppCommandLineControl.C

```diff
@@ -0,0 +1,98 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "MultiAppCommandLineControl.h"
+#include "Function.h"
+#include "Sampler.h"
+#include "MultiApp.h"
+
+registerMooseObject("StochasticToolsApp", MultiAppCommandLineControl);
+
+template <>
+InputParameters
+validParams<MultiAppCommandLineControl>()
+{
+  InputParameters params = validParams<Control>();
+  params += validParams<SamplerInterface>();
+  params.addClassDescription("Control for modifying the command line arguments of MultiApps.");
+
+  // Set and suppress the 'execute_on' flag, it doesn't work with any other flag
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_PRE_INITIAL};
+  params.suppressParameter<ExecFlagEnum>("execute_on");
+
+  params.addRequiredParam<MultiAppName>("multi_app", "The name of the MultiApp to control.");
+  params.addParam<SamplerName>(
+      "sampler",
+      "The Sampler object to utilize for altering the command line options of the MultiApp.");
+  params.addParam<std::vector<std::string>>(
+      "param_names", "The names of the command line parameters to set via the sampled data.");
+
+  return params;
+}
+
+MultiAppCommandLineControl::MultiAppCommandLineControl(const InputParameters & parameters)
+  : Control(parameters),
+    SamplerInterface(this),
+    _multi_app(_fe_problem.getMultiApp(getParam<MultiAppName>("multi_app"))),
+    _sampler(SamplerInterface::getSampler("sampler")),
+    _param_names(getParam<std::vector<std::string>>("param_names"))
+{
+  if (_multi_app->numGlobalApps() != _sampler.getTotalNumberOfRows())
+    mooseError("The number of sub apps (",
+               _multi_app->numGlobalApps(),
+               ") created by MultiApp object '",
+               _multi_app->name(),
+               "' must be equal to the number for rows (",
+               _sampler.getTotalNumberOfRows(),
+               ") for the '",
+               _sampler.name(),
+               "' Sampler object.");
+}
+
+void
+MultiAppCommandLineControl::initialSetup()
+{
+  // Do not put anything here, this method is being called after execute because the execute_on
+  // is set to PRE_INITIAL for this class. It won't work any other way.
+}
+
+void
+MultiAppCommandLineControl::execute()
```

Comment:
OK - So this is the piece of code that needs to run early. It seems like we can put this somewhere else? Perhaps in a utility and pass a bunch of things into it? I realize it's not quite that simple, just trying to spur a different line of thought to see if we can accomplish what you are after without a new execute on flag that feels misplaced. Maybe a somewhat specialized method in the Control base class for handling this special case?

Reply:
I agree the "EXEC_PRE_INITIAL" is strange; however, this is not available to any other object. The defaults are still the same. We could rename it to "_MOOSE_INTERNAL_PRE_INITIAL" or something else to indicate that it isn't a normal flag.

I used this approach because it seemed better than yet another one-of-a-kind FEProblemBase method. Also, as we move to a unified UserObject based system in a single warehouse, then these flags are the way it will be done in the future, unless I am wrong about that.

Reply:
At the point when it can only be used by one kind of system and it doesn't behave like anything else (e.g. initialSetup() won't run before the execute()), isn't another one-of-a-kind FEProblemBase method just as good (bad)? At least it's located in one spot and isn't sprinkled around in the execute flags system. 

I'm OK with renaming the flag to make it clear that it's internal (as you've proposed) but I don't see that as any better than just running a special method in FEProblemBase.

Reply:
If it is all the same to you, then I would like to leave it the way it is (with a flag rename), if for no other reason than it is done. 

There is one more reason that I recalled for my approach. The Sampler objects, which generate the random numbers, also need to execute early so the stochastic values are available to feed into the MultiApps. So rather than create two special object loops I prefer to use the system that exists for performing these loops, which will be easier to maintain as we update how we loop over objects.

Reply:
I might suggest that you use the `final` keyword in this class so that somebody else doesn't override this class and extend it in a broken state.